src/render/Renderer.cpp

```diff
@@ -331,7 +331,7 @@ bool CHyprRenderer::shouldRenderWindow(PHLWINDOW pWindow, PHLMONITOR pMonitor) {
     if (pWindow->m_pMonitor == pMonitor)
         return true;
 
-    if (!(!pWindow->m_pWorkspace || !pWindow->m_pWorkspace->isVisible()) && pWindow->m_pMonitor != pMonitor)
+    if (!(pWindow->m_pWorkspace && pWindow->m_pWorkspace->isVisible()) && pWindow->m_pMonitor != pMonitor)
```

Comment:
shouldnt this be (!workspace || !visible)?

Reply:
`!(workspace && visible)` and `(!workspace || !visible)` have the same truth table so either way works
workspace | visible | output
-- | -- | --
0 | 0 | 1
0 | 1 | 1
1 | 1 | 0
1 | 0 | 1

Reply:
right, sorry, I am a bit funky today. I'd still prefer my version as it's more clear